httpie/input.py

```diff
@@ -216,66 +216,81 @@ def _setup_standard_streams(self):
             self.env.stdout_isatty = False
 
     def _process_auth(self):
-        write_info('process_auth', 'Total: 11')
+        write_info('process_auth', 'Total: 16')
         # TODO: refactor
         self.args.auth_plugin = None
         default_auth_plugin = plugin_manager.get_auth_plugins()[0]
         auth_type_set = self.args.auth_type is not None
         url = urlsplit(self.args.url)
 
-        if self.args.auth is None and not auth_type_set:
+        if self.args.auth is None:
             has_branched('process_auth', 1)
-            if url.username is not None:
+            if not auth_type_set:
                 has_branched('process_auth', 2)
-                # Handle http://username:password@hostname/
-                username = url.username
-                password = url.password or ''
-                self.args.auth = AuthCredentials(
-                    key=username,
-                    value=password,
-                    sep=SEP_CREDENTIALS,
-                    orig=SEP_CREDENTIALS.join([username, password])
-                )
+                if url.username is not None:
+                    has_branched('process_auth', 3)
+                    # Handle http://username:password@hostname/
+                    username = url.username
+                    password = url.password or ''
+                    self.args.auth = AuthCredentials(
+                        key=username,
+                        value=password,
+                        sep=SEP_CREDENTIALS,
+                        orig=SEP_CREDENTIALS.join([username, password])
+                    )
 
         if self.args.auth is not None or auth_type_set:
-            has_branched('process_auth', 3)
-            if not self.args.auth_type:
+            # Check branch taken in compund OR above
+            if self.args.auth is not None:
                 has_branched('process_auth', 4)
+            elif auth_type_set:
+                has_branched('process_auth', 5)
+
+            if not self.args.auth_type:
+                has_branched('process_auth', 6)
                 self.args.auth_type = default_auth_plugin.auth_type
             plugin = plugin_manager.get_auth_plugin(self.args.auth_type)()
 
             if plugin.auth_require and self.args.auth is None:
```

Comment:
The "and self.args.auth is None" statement should be removed, since it has a separate if-statement below.